test/common/run-tests

```diff
@@ -278,27 +301,30 @@ def run(opts, image):
                 retry_reason, test_result = finish_test(opts, test, changed_tests)
                 result += test_result
 
-                if test is serial_test:
-                    serial_tests_duration += (time.time() - serial_test_start)
+                if test.serial_machine:
+                    tests_duration = (time.time() - batch_tests[test.serial_machine]["started"])
+                    batch_tests[test.serial_machine]["time"] += tests_duration
 
                     # sometimes our global machine gets messed up; also, tests that time out don't run cleanup handlers
                     # restart it to avoid an unbounded number of test retries and follow-up errors
                     if not opts.machine and (poll_result == 124 or (retry_reason and b"test harness" in retry_reason)):
                         # try hard to keep the test output consistent
-                        testlib.MachineCase.kill_global_machine() # TODO?
-                        testlib.MachineCase.get_global_machine()
+                        testlib.MachineCase.kill_global_machine(test.serial_machine)
+                        # TODO is the ssh adn browser address the same?
+                        testlib.MachineCase.get_global_machine(restrict=not opts.enable_network, id=test.serial_machine)
```

Comment:
@martinpitt Here we can destroy machine and get a new one. But we  generate commands with ssh_address and web_address before that. Will we get the same ports?

Reply:
You are right that this is a potential race condition if you run more than one `run-tests` at the same time on a machine. That never happens in CI though, and hopefully not too much on developer machines either. It's fine to mark this as a FIXME, but we've had this for a long time already.